### DISCUSSION TITLE
Engineering (ship overview) screen?

### DISCUSSION THREAD

**Discussion by Deadweasel** [iaincollins/icarus#12] [Ideas & Suggestions]
First off: THANK YOU for the efforts that have gone into this thing.  It's simple, fast and beautiful!  

I have been taking it for a spin doing some long-haul mining runs, and it highlighted once again that there's one aspect I think a lot of external apps like this are missing, but could benefit greatly from: an engineering overview of current ship status.
I'm talking about something that solo pilots out in the black would likely rely on, if the ED universe were reality: a general overview of the major critical things that matter at-a-glance, such as fuel level, cargo level, shield strength, hull strength, deployed scoops/hardpoints/gear (because trying to remember what's sticking out if you have to panic-jump can be maddening!), failing/damaged modules, etc.

I know a few of these things are present already in the Ship tab primary view, but I'm talking about a bolder, more comprehensive graphical view that will allow the pilot to see the general status of things with color-coding or progress bars, the kind of thing I could have up on my sidekick display that I would have to rummage through multiple screens/tabs or tiny UI indicators in the game to see otherwise.  It would be really nice to be able to glance over at the screen, see the general state of things, then request docking and see the display shift to present where the dang pad is in the station.  :)

That said, I do really like the current ship modules overview as it currently exists.  Makes it nice to see a little more easily what modules haven't been engineered to hell and back yet!  The rest of the whole interface fulfills a lot of dreams for me too, especially the navigation routing where I can see my progress to the current planned destination!
An oddity I noted with that view too: if the route is changed after initial plot, Icarus doesn't seem to pick up the changes.  Say I plot to a destination 18 hops away, then realize I forgot the limpits (d'oh!), so load up, and now the route updates to accommodate the new weight.  The game of course updates, but Icarus doesn't seem to catch that.  Do I need to clear and re-plot the course from scratch to force the issue for it?

**Comment by iaincollins** [comment]
Hi there!

Thank you, I'm glad you like it!

> I'm talking about something that solo pilots out in the black would likely rely on, if the ED universe were reality: a general overview of the major critical things that matter at-a-glance, such as fuel level, cargo level, shield strength, hull strength, deployed scoops/hardpoints/gear (because trying to remember what's sticking out if you have to panic-jump can be maddening!), failing/damaged modules, etc.

> I know a few of these things are present already in the Ship tab primary view, but I'm talking about a bolder, more comprehensive graphical view that will allow the pilot to see the general status of things with color-coding or progress bars, the kind of thing I could have up on my sidekick display 

Absolutely - improvements to tie in module and engineering (e.g. recommendations, other options, details about experimental effects) are definitely planned! A real time graphical view of the ship status is my current focus and I'm really excited about getting that done. 

The Ship panel will be getting a huge overhaul; to the point the module view may end up becoming it's own view under the Ship panel, that just focuses on modules. I'm unclear if I'll do a huge update in one go or a series of smaller updates. It might depend on if I do a custom panel for each ship - which I would love to do and would be really great but a lot more work.

I'd also like to explore making the Ship view interactive so you can preform actions (e.g. toggle hardpoints, lights, landing gear, pips, fighter commands, etc) although I anticipate the first iteration being a read only view.

The only sticking point I ran into has been that the game API isn't great for getting info about damaged/failing modules, which is a sham. That was actually in earlier releases, but I ended up removing as the game doesn't expose access to real time data for this which makes it not so useful; unlike for other data, information exposed via the API about module state gets written quite infrequently, under specific circumstances and can be several minutes old.


> An oddity I noted with that view too: if the route is changed after initial plot, Icarus doesn't seem to pick up the changes. Say I plot to a destination 18 hops away, then realize I forgot the limpits (d'oh!), so load up, and now the route updates to accommodate the new weight. The game of course updates, but Icarus doesn't seem to catch that. Do I need to clear and re-plot the course from scratch to force the issue for it?

Hmm interesting, I haven't run into that but it's possibly a bug with the game API. ICARUS updates in real time whenever you plot a new course but maybe when the game is handling route planning it's not writing updated routes out via the API. Opening the Galaxy Map in game can sometimes fix this.

**Reply by Deadweasel** [reply]
Okay now I'm definitely stoked to see what's coming down the pipe!

Thanks so much for the update, and again, for making such an awesome, fluid app that feels exactly like it belongs with the game!
